File: src/main/go/leetcode/lists/offer/Offer15.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2022/12/28 21:49
 */

func hammingWeight(num uint32) int {
	count := 0
	for i := 0; i < 32; i++ {
		if (num & 1) == 1 {
			count++
		}
		num >>= 1
	}
	return count
}

File: java/src/leetcode/lists/offer/Offer15.java
package leetcode.lists.offer;

/**
 * @author wujingxinit@outlook.com
 * @date 2023/9/12 15:50
 */
public class Offer15 {
    public int hammingWeight(int n) {
        int ret = 0;
        while (n != 0) {
            n &= n - 1;
            ret++;
        }
        return ret;
    }
}
